### ISSUE TITLE
Container is not ready

### ISSUE THREAD

**Issue by mtparet** [aylei/kubectl-debug#52]
Hello there,

We are trying to debug a pod which failed to correctly startup in some conditions, only on production. But when launching the debug command, it failed because the container of the application is not ready.

```
kubectl debug app-XXX -n production --agentless
Agent Pod info: [Name:debug-agent-pod-f0fb8c3c-d083-11e9-844f-9cb6d0eeb5ef, Namespace:default, Image:aylei/debug-agent:latest, HostPort:10027, ContainerPort:10027]
Waiting for pod debug-agent-pod-f0fb8c3c-d083-11e9-844f-9cb6d0eeb5ef to run...
error: container [app-XXX] not ready
```

Does this condition is needed to debug pod ?

Best,

Matthieu

**Comment by aylei** [comment]
This is expected and `--fork` option was introduced to tackle such problems. With `--fork` option, a copy of the target pod will be created with the start command being replaced to `tail -f /dev/null`. Then you can run the real application binary from the debug container to investigate more.

see: https://github.com/aylei/kubectl-debug#debug-instructions

**Comment by mtparet** [comment]
@aylei thanks for this first answer.

I understand the point when the pod is not running (or terminated / crashed ) but here the pod is running but it is just not marked as ready.

We have an issue happening only when we got external traffics to it, the application freeze (and so the pod becomes not ready which is expected) and this is at this moment we'll like to use kubectl debug.

**Comment by aylei** [comment]
Interesting case, so there is a readinessProbe configured I suppose? Seems like that kubectl-debug should establish connection to Pod despite the `NotReady` status then.

**Comment by mtparet** [comment]
Indeed we have `readinessProbe` configured to ensure the web server is responding. 

> Seems like that kubectl-debug should establish connection to Pod despite the NotReady status then.

If we can, that would be perfect for this use case.

**Comment by mtparet** [comment]
@aylei I would like to help on this, do you have any advice on how we could allow debug NotReady pod ?

**Comment by aylei** [comment]
@mtparet That'd be great! https://github.com/aylei/kubectl-debug/blob/8fbb788cce08988bf14112e54f578b3017a25e27/pkg/plugin/cmd.go#L491 The contaienr in `NotReady` state is filterer our here, we can relax this constraint.

**Comment by multi-io** [comment]
@mtparet are you still working on this? Otherwise we can submit the PR too (if it's just that one conditional). This is a pretty annoying thing, would like to have this fixed :)

**Comment by calmkart** [comment]
fixed in [pull request#62](https://github.com/aylei/kubectl-debug/pull/62)
